src/components/PromisesTable.vue

```diff
@@ -79,14 +93,48 @@ export default {
     exclude: { type: Array, default: () => [] }
   },
   data: () => ({
-    search: ''
+    search: '',
+    filterStatus : '',
+    statusOptions: [
+    {
+      value: '',
+      label: 'All'
+    }, {
+      value: 'Review Needed',
+      label: 'Review Needed'
+    }, {
+      value: 'Not Started',
+      label: 'Not Started'
+    }, {
+      value: 'In Progress',
+      label: 'In Progress'
+    }, {
+      value: 'Fulfilled',
+      label: 'Fulfilled'
+    }, {
+      value: 'Partially Fulfilled',
+      label: 'Partially Fulfilled'
+    }, {
+      value: 'At Risk',
+      label: 'At Risk'
+    }, {
+      value: 'Retracted',
+      label: 'Retracted'
+    }, {
+      value: 'Broken',
+      label: 'Broken'
+    }],
   }),
   methods: { formatDate },
   computed: {
     filteredPromises () {
-      return this.promises.filter(promise => {
+      var results = this.promises.filter(promise => {
         return promise.title.toLowerCase().includes(this.search.toLowerCase())
       })
+      results = results.filter(promise => {
```

Comment:
best to define the filtering in its own method like how search is being done in the line above :)